File: python/프로그래머스/level-1/체육복.py
def solution(n, lost, reserve):
    # 여벌 체육복을 가져온 학생이 체육복을 도난당했을 수 있습니다.
    # -> 차집합을 구함
    set_reserve = set(reserve) - set(lost)
    set_lost= set(lost) - set(reserve)
    
    for v in set_reserve:
        # 왼쪽 사람부터 줘야 최댓값을 구할 수 있다.
        if v-1 in set_lost:
            set_lost.remove(v-1)
        elif v+1 in set_lost:
            set_lost.remove(v+1)
        
    return n-len(set_lost)

File: java/프로그래머스/level-1/체육복.java
import java.util.Arrays;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
       int answer = n - lost.length;
        
        Arrays.sort(lost);
        Arrays.sort(reserve);
        
        // 여벌의 체율복을 가지고 있다 도난당한 경우
        for (int i =0; i <lost.length; i ++){
            for (int j =0; j <reserve.length; j ++){
                 if(lost[i] == reserve[j]){
                    answer++;
                     lost[i] = -1;
                     reserve[j]= -1;
                     break;
                 }
        
        }   
        }
        
        // 도난당한 학생에게 체육복 빌려주는 경우
        for (int i =0; i <lost.length; i ++){
            for (int j =0; j <reserve.length; j ++){
            if ((lost[i]-1 == reserve[j]) || (lost[i]+1 == reserve[j])){
            answer ++;
            reserve[j]= -1;    
            break;
            }
            
            
        }   
        }
        return answer;
    }
}